torch/distributed/optim/zero_redundancy_optimizer.py

```diff
@@ -98,8 +102,6 @@ class ZeroRedundancyOptimizer(Optimizer):
     Keyword Args:
         optim (torch.nn.Optimizer): optimizer to shard
         group (group): torch.distributed group (default: group.WORLD)
-        bucket_cap (int): the size of the buffer used to batch the small parameter tensors,
```

Comment:
yes sorry, I should have added an explanation here: basically with these tensor views, there's no need for bucketing really, memory wise the parameters are part of an optimal big bucket, but from the outside they're exposed as normal parameters (ie: they're tensor views of the bucket). When the shards are sent (broadcast/all-gather), the "background" buffer is synced in between the ranks, so there's one call per rank<>rank, instead of a lot of small calls.

To add more context and compare with DDP/ShardedDDP reduce buckets, it's not the same situation really, because in that case we don't wipe anything over time and there's no latency cost in using buckets. The parameters are always 100% there (the update process is sharded, but not the params as such), there's no drawback in having maximal "tensor view buckets" really, it does not take any extra space and it just removes calls. In the case of reduce with DDP the buckets are more of a tradeoff, for instance because you need to wait for them to fill up, so it makes sense to expose the knob I think. 

There's no overlap currently in between the broadcast and the FW, and frankly it's killing performance, if there's a way you can think of that would be game changing, I spent some time on that, outside of breaking the model in subparts (what FSDP does) I could not find anything. Within a node it's no problem, but in between nodes with slow interconnect it's easily the bottleneck.